src/Config/RuntimeConfigLoader.cs

```diff
@@ -112,9 +124,16 @@ public static JsonSerializerOptions GetSerializationOptions()
         options.Converters.Add(new EnumMemberJsonEnumConverterFactory());
         options.Converters.Add(new RestRuntimeOptionsConverterFactory());
         options.Converters.Add(new GraphQLRuntimeOptionsConverterFactory());
-        options.Converters.Add(new EntitySourceConverterFactory());
+        options.Converters.Add(new EntitySourceConverterFactory(replaceEnvVar));
+        options.Converters.Add(new EntityGraphQLOptionsConverterFactory(replaceEnvVar));
+        options.Converters.Add(new EntityRestOptionsConverterFactory(replaceEnvVar));
```

Comment:
I think factory pattern just fits nicely here.

The CLI is going to be using the same `TryLoadKnownConfig` function that comes from the `FileSystemRuntimeConfigLoader`, but it has no reason to do the environmental variable parsing in most cases. The factory pattern here lets the CLI generate the right derived class that has the behavior it wants with as you noticed a super simple addition of simply passing in this flag. So with that flag we get the derived class with the deserialization behavior that won't replace all those env variables.

It could be done other ways, but this is pattern that can slot into the existing code nicely in my opinion.

Reply:
When I did the original implementation, the reason that some used factories and others didn't was because there were some instances where we would want to remove the converter dynamically. This would happen when the fully expanded .NET type mapped directly to the JSON structure (when not using a shorthand), and for those cases, rather than walking through the JSON Reader it'd just call into the deserialization engine again. The only problem would be that if the converter was still there, it would become a recursive call, eventually resulting in exceeding the call stack depth.

The only workaround for this is to "manually" add the converter with a factory rather than relying on the attribute-discovery, as if it's discovered by an attribute you can't remove it dynamically.

Reply:
And the reason I added the factory is yes, because I wanted to pass on `replaceEnvVar` to the constructor otherwise there was chance the default constructor would be picked up. Even if the right constructor is invoked, we wouldn't be able to pass in the argument of the `GetSerializationOptions` function downstream to the constructor that would be invoked.